bitcoin_executor/sources/ripemd160.move

```diff
@@ -0,0 +1,470 @@
+        h.bytes = h.bytes + 64 - bufsize;
+        data_index = data_index + 64 - bufsize;
+        transform(&mut h.s, h.buf);
+        bufsize = 0;
+    };
+
+    while (end - data_index >= 64) {
+        let mut v: vector<u8> = vector[];
+        do!(64, |i| v.push_back(data[i + data_index]));
+        transform(&mut h.s, v);
+        h.bytes = h.bytes + 64;
+        data_index = data_index + 64;
+    };
+
+    if(end > data_index) {
+        veccopy(&mut h.buf, bufsize, data, data_index,end-data_index);
+        h.bytes = h.bytes + end - data_index;
+    };
+}
+
+public fun finalize(h: &mut Ripemd160): vector<u8> {
+    let mut pad: vector<u8> = vector[0x80];
+    let mut i = 1;
+    while (i < 64) {
+        pad.push_back(0);
+        i = i + 1;
+    };
+
+    let bytes = h.bytes;
+
+    let mut sizedecs: vector<u8> = vector[0, 0, 0, 0, 0, 0, 0, 0];
+    writeLE64(&mut sizedecs, 0, bytes << 3);
+    h.write(pad, 1 + ((119 - (bytes % 64)) % 64));
+    h.write(sizedecs, 8);
+    let mut hash: vector<u8> = vector[];
+    i = 0;
+    while (i < 20) {
+        hash.push_back(0);
+        i = i + 1;
+    };
+
+    writeLE32(&mut hash, 0, h.s[0]);
+    writeLE32(&mut hash, 4, h.s[1]);
+    writeLE32(&mut hash, 8, h.s[2]);
+    writeLE32(&mut hash, 12, h.s[3]);
+    writeLE32(&mut hash, 16, h.s[4]);
+
+    hash
+}
+
+fun veccopy(dest: &mut vector<u8>, dest_start: u64, src: vector<u8>, src_start: u64, len: u64) {
+    let mut i = dest_start;
+    let mut j = src_start;
+    let mut k = 0;
+    while (k < len) {
+        let b = dest.borrow_mut(i);
+        *b = src[j];
+        i = i + 1;
+        j = j + 1;
+        k = k + 1;
+    }
+}
+
+fun writeLE64(v: &mut vector<u8>, start_index: u64, x: u64) {
+    let mut i = 0;
+    let mut x = x;
+    let mut index = start_index;
+    while (i < 8) { // 64 bits
+        let b = v.borrow_mut(index);
+        *b = (x % 256) as u8;
+        x = x / 256;
+        index = index + 1;
+        i = i + 1;
+    }
+}
+
+fun writeLE32(v: &mut vector<u8>, start_index: u64, x: u32) {
+    let mut i = 0;
+    let mut x = x;
+    let mut index = start_index;
+    while (i < 4) { // 64 bits
+
+        let b = v.borrow_mut(index);
+        *b = (x % 256) as u8;
+        x = x / 256;
+        i = i + 1;
+        index = index + 1;
+    }
+}
+fun readLE32(v: &vector<u8>, start_index: u64): u32 {
+    let mut ans = 0;
+    let mut start_index = start_index;
+    let mut base = 1;
+    let mut i = 0;
+    while (i < 4) {
+        ans = ans + base * (v[start_index] as u32);
+        if (i == 3) {
+            break
+        };
+        base = base * 256;
+        i = i + 1;
+        start_index = start_index + 1;
+    };
+    ans
+}
+
+#[test]
+fun ripemd160_test() {
+    let data: vector<u8> = std::hash::sha2_256(x"0102");
+    let mut h = new();
+    h.write(data, data.length());
+    let hash = h.finalize();
+    sui::test_utils::assert_eq(hash, x"15cc49e191cbc520d91944600a5cb77af6aa3291");
+}
```

Comment:
please add more tests. You can generate data and correct answers using python or JS. Let's add at least:
* hashing data with 1 byte
* hasing data with 2 bytes
* hasing data with 100 bytes
* hashing data with 10kB
* hashing `0x00`
* hashing `0x00000000`

Reply:
Added task for this!